pkg/cli/cmd/bicep/generatekubernetesmanifest/generatekubernetesmanifest_test.go

```diff
@@ -0,0 +1,699 @@
+			},
+			ValidateCallback: func(t *testing.T, r framework.Runner) {
+				runner := r.(*Runner)
+				require.Equal(t, "aws-scope-value", runner.AWSScope)
+				require.Equal(t, "/planes/radius/local/resourceGroups/default", runner.DeploymentScope)
+			},
+		},
+		{
+			Name:          "rad bicep generate-kubernetes-manifest - valid with deployment scope",
+			Input:         []string{"app.bicep", "--deployment-scope", "deployment-scope-value"},
+			ExpectedValid: true,
+			ConfigHolder: framework.ConfigHolder{
+				ConfigFilePath: "",
+				Config:         configWithWorkspace,
+			},
+			ValidateCallback: func(t *testing.T, r framework.Runner) {
+				runner := r.(*Runner)
+				require.Equal(t, "deployment-scope-value", runner.DeploymentScope)
+			},
+		},
+		{
+			Name:          "rad bicep generate-kubernetes-manifest - missing file argument",
+			Input:         []string{},
+			ExpectedValid: false,
+			ConfigHolder: framework.ConfigHolder{
+				ConfigFilePath: "",
+				Config:         radcli.LoadEmptyConfig(t),
+			},
+		},
+	}
+
+	radcli.SharedValidateValidation(t, NewCommand, testcases)
+}
+
+func Test_Run(t *testing.T) {
+	t.Run("Create DeploymentTemplate (basic)", func(t *testing.T) {
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		template := `
+		{
+      "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+      "contentVersion": "1.0.0.0",
+      "imports": {
+        "Radius": {
+          "provider": "Radius",
+          "version": "latest"
+        }
+      },
+      "languageVersion": "2.1-experimental",
+      "metadata": {
+        "_EXPERIMENTAL_FEATURES_ENABLED": [
+          "Extensibility"
+        ],
+        "_EXPERIMENTAL_WARNING": "This template uses ARM features that are experimental. Experimental features should be enabled for testing purposes only, as there are no guarantees about the quality or stability of these features. Do not enable these settings for any production usage, or your production environment may be subject to breaking.",
+        "_generator": {
+          "name": "bicep",
+          "templateHash": "10886769892319697000",
+          "version": "0.30.23.60470"
+        }
+      },
+      "resources": {
+        "basic": {
+          "import": "Radius",
+          "properties": {
+            "name": "basic",
+            "properties": {
+              "compute": {
+                "kind": "kubernetes",
+                "namespace": "default",
+                "resourceId": "self"
+              },
+              "recipes": {
+                "Applications.Datastores/redisCaches": {
+                  "default": {
+                    "templateKind": "bicep",
+                    "templatePath": "ghcr.io/radius-project/recipes/local-dev/rediscaches:latest"
+                  }
+                }
+              }
+            }
+          },
+          "type": "Applications.Core/environments@2023-10-01-preview"
+        }
+      }
+    }
+		`
```

Comment:
Can this moved into a file inside `testdata/`? It's kinda ugly to maintain JSON inside Go and there's an easy solution.

Reply:
fixed